### PR TITLE
Add /role.all, get, create, update, delete to Admin API

### PR DESCRIPTION
**Pull request by unnawut**
Issue/Task Number: #490
Required by #490

# Overview

This PR implements the role endpoints (all, get, create, update, delete) required to continue with the actual permissions in #490.

This is a separate PR from the rest of the permissions to make the review easier.

# Changes

- Added controllers and supporting modules for /role.all, /role.get, /role.create, /role.update, /role.delete
- Refactored `Role.get_by_name/1` into `Role.get_by/2`
- Added external ID and soft deletion to `Role` schema
- Added OpenAPI specs for the new endpoints

# Implementation Details

Straightforward adding of endpoints to retrieve and manage roles. The `Role` schema did not have external ID nor soft deletion before, but to keep the endpoints and their usage consistent, it makes sense to add them so implementers can expect consistent behavior across the eWallet.

# Usage

Retrieve all roles:

```sh
curl 10.5.10.10:4000/api/admin/role.all \
-H "Accept: application/vnd.omisego.v1+json" \
-H "Authorization: OMGAdmin <truncated>" \
-H "Content-Type: application/json" \
-d '{
}' \
-v -w "\n"
```

Update a role:

```sh
curl 10.5.10.10:4000/api/admin/role.update \
-H "Accept: application/vnd.omisego.v1+json" \
-H "Authorization: OMGAdmin <truncated>" \
-H "Content-Type: application/json" \
-d '{
  "id": "rol_01cw625zm2xmny5ee3mwt9j8ky",
  "name": "new_name"
}' \
-v -w "\n"
```

# Impact

DB migration and new API endpoints

### CODE REVIEW

File: apps/admin_api/lib/admin_api/v1/controllers/role_controller.ex [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+  end
+
+  def update(conn, _), do: handle_error(conn, :invalid_parameter)
+
+  @doc """
+  Soft-deletes an existing role by its id.
+  """
+  @spec delete(Plug.Conn.t(), map()) :: Plug.Conn.t()
+  def delete(conn, %{"id" => id} = attrs) do
+    with %Role{} = role <- Role.get(id) || {:error, :role_id_not_found},
```
Comment (mederic-p) [inline apps/admin_api/lib/admin_api/v1/controllers/role_controller.ex]: Shouldn't we have a `permit` here too?

Comment (unnawut) [reply apps/admin_api/lib/admin_api/v1/controllers/role_controller.ex]: Updated. Please check again.

File: apps/admin_api/test/admin_api/v1/controllers/admin_auth/role_controller_test.exs [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+
+      response = admin_user_request("/role.all", attrs)
+      roles = response["data"]["data"]
+
+      assert response["success"]
+      assert Enum.count(roles) == 3
+      assert Enum.at(roles, 0)["name"] == "matched_3"
+      assert Enum.at(roles, 1)["name"] == "matched_2"
+      assert Enum.at(roles, 2)["name"] == "matched_1"
+    end
```
Comment (T-Dnzt) [inline apps/admin_api/test/admin_api/v1/controllers/admin_auth/role_controller_test.exs]: We should start adding a test that uses the match parsers for all the `.all` endpoints.

File: apps/ewallet/lib/ewallet/policies/role_policy.ex [hunk 3]
```diff
@@ -0,0 +1,23 @@
+defmodule EWallet.RolePolicy do
+  @moduledoc """
+  The authorization policy for roles.
+  """
+  @behaviour Bodyguard.Policy
+  alias EWalletDB.{Account, User}
+
+  # Any user can get a role
+  def authorize(:all, _user_or_key, _role_id), do: true
```
Comment (T-Dnzt) [inline apps/ewallet/lib/ewallet/policies/role_policy.ex]: Let's be more explicit here and use `admin_user:` or `key: `

File: apps/ewallet/lib/ewallet/web/v1/serializers/role_serializer.ex [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+
+  def serialize(%Role{} = role) do
+    %{
+      object: "role",
+      id: role.id,
+      name: role.name,
+      priority: role.priority,
+      display_name: role.display_name,
+      user_ids: UserSerializer.serialize(role.users, :id),
+      users: UserSerializer.serialize(role.users),
```
Comment (T-Dnzt) [inline apps/ewallet/lib/ewallet/web/v1/serializers/role_serializer.ex]: Do we really want this? I don't think we should send back users here na khrap.

File: apps/admin_api/test/admin_api/v1/controllers/admin_auth/role_controller_test.exs (hunk 2 above)
Comment (unnawut) [reply apps/admin_api/test/admin_api/v1/controllers/admin_auth/role_controller_test.exs]: Updated. Please check.

File: apps/ewallet/lib/ewallet/policies/role_policy.ex (hunk 3 above)
Comment (unnawut) [reply apps/ewallet/lib/ewallet/policies/role_policy.ex]: Updated. Please check.

File: apps/ewallet/lib/ewallet/web/v1/serializers/role_serializer.ex (hunk 4 above)
Comment (unnawut) [reply apps/ewallet/lib/ewallet/web/v1/serializers/role_serializer.ex]: Updated. Please check.

File: apps/admin_api/test/support/conn_case.ex [hunk 5]
```diff
@@ -309,4 +315,105 @@ defmodule AdminAPI.ConnCase do
     insert_list(num_remaining, factory_name, attrs)
     Repo.all(schema)
   end
+
+  @doc """
+  Tests that the specified endpoint supports 'match_any' filtering.
+  """
+  defmacro test_supports_match_any(endpoint, auth_type, factory, field, opts \\ []) do
```
Comment (T-Dnzt) [inline apps/admin_api/test/support/conn_case.ex]: 👍

**Pull request merged**